smart-contracts/contracts/DigitalaxAuction.sol

```diff
@@ -89,23 +139,18 @@ contract DigitalaxAuction is Context, ReentrancyGuard {
         uint256 _reservePrice,
         uint256 _startTime,
         uint256 _endTime
-    // TODO may need to take in lister is setup is done from a orchestrating contract
     ) external {
         // Ensure caller has privileges
-        require(
-        // TODO change this to smart contract check assuming something else is orchestrating it
-            accessControls.hasMinterRole(_msgSender()),
-            "DigitalaxAuction.createAuction: Sender must have the minter role"
-        );
+        require(accessControls.hasMinterRole(_msgSender()), "DigitalaxAuction.createAuction: Sender must have the minter role");
 
         // Check end time not before start time
         require(_endTime > _startTime, "DigitalaxAuction.createAuction: End time must be greater than start");
 
         // Check another auction not already flight
-        require(
-            _getNow() > auctions[_garmentTokenId].endTime,
-            "DigitalaxAuction.createAuction: Cannot create an auction in the middle of another"
-        );
+        require(_getNow() > auctions[_garmentTokenId].endTime, "DigitalaxAuction.createAuction: Cannot create an auction in the middle of another");
+
+        // Check owner of the token is the creator
+        require(garmentNft.ownerOf(_garmentTokenId) == _msgSender(), "DigitalaxAuction.createAuction: Cannot create an auction if you dont own it");
```

Comment:
`don't` and I would put `don't own garment`

Reply:
or `do not`

Reply:
will adjust